frontend/src/lib/pipelineModes.ts

```diff
@@ -0,0 +1,108 @@
+import { PIPELINES } from "../data/pipelines";
+import { getCachedPipelineSchema } from "./utils";
+import { GENERATION_MODE, type GenerationMode } from "../constants/modes";
+import type { PipelineCategory, PipelineId } from "../types";
+
+export interface PipelineModeCapabilities {
```

Comment:
I'd focus on the fields we have in this interface. I think a lot of properties here are either redundant or unused. E.g. 1. `requiresVideoInVideoMode`, isn't it the same as `supportsVideo`? I think we just have Video mode and Text mode
2. `category` => Isn't it the same as these `supportsVideo`, `supportsText`?

In general, maybe we could try to reduce the number of properties to what we need.